src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java

```diff
@@ -424,15 +427,20 @@ public Resource get(String resourcePath) {
 
         @Override
         public Type getType() {
-            if (!file.exists()) {
+            try {
+                BasicFileAttributes attributes = java.nio.file.Files.readAttributes(file.toPath(), BasicFileAttributes.class);
```

Comment:
On a cold file system this single change makes the whole startup 3 times faster (compared to a version with everything else, without this one change). . On a hot one, it helps but nowhere near as much, the other changes have more weight.